docs/guides/using-org-slugs.mdx

```diff
@@ -0,0 +1,397 @@
+  > `afterSelectOrganizationUrl`.
+
+  ### Using clerkMiddleware to set the active organization
+
+  With [clerkMiddleware](https://clerk.com/docs/references/nextjs/clerk-middleware), you can
+  declare URL patterns that indicate that a specific organization or the user's [personal account](https://clerk.com/glossary#personal-account)
+  should be activated. If the middleware detects one
+  of these patterns in the URL, and it notices that a different organization is
+  currently active according to the session, it will activate the indicated organization if
+  possible.
+
+  > [!NOTE]
+  > Implementation Detail: To resolve a mismatch, Clerk middleware initiaties a [Handshake](docs/references/sdk/terminology#:~:text=each:Handshake).
+  > The middleware will respond to the browser with a redirect to Clerk's API with a request to activate the URL-indicated
+  > organization. If it's possible to do so, Clerk's API will respond with a redirect back to the original URL, including
+  > a session token with the desired organization as active and redirect back to your application.
+
+  For this application, the path `/orgs/:slug` with any optional trailing path segments should activate the
+  organization indicated by the slug. To achieve this, you can define two `organizationPatterns`: one for the
+  root (e.g. `/orgs/acmecorp`), and one using the wildcard matcher `(.*)` that will match
+  `/orgs/acmecorp/any/other/resource`.
+
+  You can use the same approach for the [personal account](https://clerk.com/glossary#personal-account) with `personalAccountPatterns`.
+
+  > [!WARNING]
+  > It may not always be possible to activate the organization requested in the URL. This can happen if no organization
+  > with the given slug exists, or the user isn't a member of the given organization. In these cases, clerkMiddleware
+  > **will not modify** the active organization, leaving as active whichever organization was previously active on the
+  > clerk session.
+
+  See the documentation for [organizationSyncOptions](/docs/references/nextjs/clerk-middleware#organization-sync-options)
+  for more specific usage of the middleware options.
+
+  ```tsx {{ filename: 'middleware.ts' }}
+  import { clerkMiddleware } from '@clerk/nextjs/server'
+
+  export default clerkMiddleware(
+    (auth, req) => {
+      if (isProtectedRoute(req)) auth().protect()
+    },
+    {
+      organizationSyncOptions: {
+        organizationPatterns: ['/orgs/:slug', '/orgs/:slug/(.*)'],
+        personalAccountPatterns: ['/me', '/me/(.*)'],
+      },
+    },
+  )
+
+  export const config = {
+    matcher: [
+      // Skip Next.js internals and all static files, unless found in search params
+      '/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)',
+      // Always run for API routes
+      '/(api|trpc)(.*)',
+    ],
+  }
```

Comment:
We're not defining `isProtectedRoute`, and I also think we can skip the middleware checks here to highlight `organizationSyncOptions`:
```suggestion
  ```tsx {{ filename: 'middleware.ts' }}
  import { clerkMiddleware } from '@clerk/nextjs/server'

  export default clerkMiddleware(
    (auth, req) => {
         // Add your middleware checks
    },
    {
      organizationSyncOptions: {
        organizationPatterns: ['/orgs/:slug', '/orgs/:slug/(.*)'],
        personalAccountPatterns: ['/me', '/me/(.*)'],
      },
    },
  )

  export const config = {
    matcher: [
      // Skip Next.js internals and all static files, unless found in search params
      '/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)',
      // Always run for API routes
      '/(api|trpc)(.*)',
    ],
  }
```